stream_alert/rule_processor/threat_intel.py

```diff
@@ -411,11 +416,42 @@ def normalized_type_mapping(cls):
         return cls.__normalized_types
 
     @staticmethod
-    def is_public_ip(ip_address):
+    def valid_ip(ip_address):
+        """Verify that a ip_address string is valid
+
+        Args:
+            ip_address (string): address to be tested
+
+        Returns:
+            True if the ip_address is valid, otherwise False
+        """
+
         try:
-            ip_addr = IPAddress(str(ip_address))
-            # We also need to filter multicast ip which is a private ip. For example,
-            # multicast ip '239.192.0.1', is a private ip.
-            return ip_addr.is_unicast() and not ip_addr.is_private()
-        except: # pylint: disable=bare-except
+            IPAddress(ip_address)
+        except AddrFormatError:
+            LOGGER.error('IP address failed parse: %s', ip_address)
             return False
+        return True
+
+    def is_excluded_ioc(self, ioc_type, ioc_value):
+        """
+        check if we should bypass IOC lookup for specified IOC
+        Args:
+            ioc_type (string): the type of IOC to evaluate (md5, ip, domain)
+            value (string): the value of IOC to evaluate
+
+        Returns:
+            True if IOC lookup should be bypassed for this value
+            False if IOC should be looked up
+        """
+        if ioc_type == 'ip':
+            try:
+                ip_addr = IPAddress(str(ioc_value))
+                excluded_networks = self.excluded_iocs.get('ip', set())
+                return in_network(ip_addr, excluded_networks)
+            except: # pylint: disable=bare-except
+                LOGGER.error('IP IOC Exclusion Failed: %s', ioc_value)
```

Comment:
Please use `LOGGER.exception(...)` here so we know what the error was if we need to troubleshoot

Reply:
this one's trickier, most of the failures are from when the source ip is not in fact an ip (it's an amazon domain). i think this might cause a lot of exceptions, though i could write code to short circuit it out before it reaches this stage

Reply:
yeah, i think it would be noisy unless we dump the domain "ips" earlier